libraries/libfc/include/fc/container/ordered_diff.hpp

```diff
@@ -36,50 +41,50 @@ class ordered_diff {
 
       diff_result result;
       while (s < source.size() || t < target.size()) {
+         FC_ASSERT(s < std::numeric_limits<SizeType>::max());
+         FC_ASSERT(t < std::numeric_limits<SizeType>::max());
          if (s < source.size() && t < target.size()) {
             if (source[s] == target[t]) {
                // nothing to do, skip over
-               assert(s <= std::numeric_limits<SizeType>::max());
-               assert(t <= std::numeric_limits<SizeType>::max());
                ++s;
                ++t;
             } else { // not equal
                if (s == source.size() - 1 && t == target.size() - 1) {
+                  FC_ASSERT(result.remove_indexes.size() < MAX_NUM_ARRAY_ELEMENTS);
+                  FC_ASSERT(result.insert_indexes.size() < MAX_NUM_ARRAY_ELEMENTS);
```

Comment:
I'd be tempted to also move these two right after the `while` so they are not repeated throughout.